### PR TITLE
LibJS: Make sure loop results are initialized

### PR DESCRIPTION
**Pull request by gunnarbeutner**
This ensures that "while", do...while, "for" and scope expressions have a properly initialized result value even if the user terminated the loop body via break or the loop body wasn't executed at all.

### CODE REVIEW

**Comment by ryangjchandler** [comment]
I've covered this in #7915, not sure which one should handle it though.

File: Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp [hunk 1]
```diff
@@ -26,6 +26,10 @@ Optional<Bytecode::Register> ScopeNode::generate_bytecode(Bytecode::Generator& g
     for (auto& child : children()) {
         last_value_reg = child.generate_bytecode(generator);
     }
+    if (!last_value_reg.has_value()) {
+        last_value_reg = generator.allocate_register();
+        generator.emit<Bytecode::Op::Load>(*last_value_reg, js_undefined());
+    }
```
Comment (linusg) [inline Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp:32]: This seems wrong. Why would a ScopeNode set its last value to undefined? This is used for BlockStatement AFAICT, see https://tc39.es/ecma262/#sec-block-runtime-semantics-evaluation.

Comment (gunnarbeutner) [reply Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp]: And it's gone. :)

**Comment by linusg** [comment]
Conflict...

**Comment by gunnarbeutner** [comment]
Rebased, or rather - rewritten. :partying_face:

**Pull request merged**